### PR TITLE
docs: update docker-compose 

### PR DESCRIPTION
**Pull request by nytai**
### SUMMARY

Update docker-compose docs to emphasize the issues with the `docker-compose up` and the node assets not building for 95% of users. Also updating the suggested tag to the latest release. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: docs/docs/installation/installing-superset-using-docker-compose.mdx [hunk 1]
```diff
@@ -75,9 +78,9 @@ the branch/tag, and then starting `docker-compose` with the `TAG` variable.
 For example, to run the 2.1.0 version, run the following commands:
```
Comment (justinpark) [inline docs/docs/installation/installing-superset-using-docker-compose.mdx:78]: + 1
```suggestion
For example, to run the 3.0.0 version, run the following commands:
```

File: docs/docs/installation/installing-superset-using-docker-compose.mdx [hunk 2]
```diff
@@ -72,12 +75,12 @@ docker-compose -f docker-compose-non-dev.yml up
...
-For example, to run the 2.1.0 version, run the following commands:
+For example, to run the 3.0.0 version, run the following commands:
 
 ```bash
-git checkout 2.1.0
-TAG=2.1.0 docker-compose -f docker-compose-non-dev.yml pull
-TAG=2.1.0 docker-compose -f docker-compose-non-dev.yml up
+git checkout 3.0.0
+TAG=3.0.0 docker-compose -f docker-compose-non-dev.yml pull
+TAG=0.0.0 docker-compose -f docker-compose-non-dev.yml up
```
Comment (sfirke) [inline docs/docs/installation/installing-superset-using-docker-compose.mdx:83]:
```suggestion
TAG=3.0.0 docker-compose -f docker-compose-non-dev.yml up
```

File: docs/docs/installation/installing-superset-using-docker-compose.mdx [hunk 3]
```diff
@@ -62,6 +62,9 @@ When working on master branch, run the following commands to run `development` m
 ```bash
 docker compose up
 ```
+:::tip
+When running in development mode the `superset-node` container needs to finish building assets in order for the UI to render properly. If you would just like to try out Superset without making any code changes follow the steps documented for `production` or a specific version bellow.
```
Comment (sfirke) [inline docs/docs/installation/installing-superset-using-docker-compose.mdx:66]:
```suggestion
When running in development mode the `superset-node` container needs to finish building assets in order for the UI to render properly. If you would just like to try out Superset without making any code changes follow the steps documented for `production` or a specific version below.
```

**Review by sfirke — CHANGES_REQUESTED** [review]

**Review by sfirke — APPROVED** [review]
LGTM

**Pull request merged**